cmd/example-mysql/main.go

```diff
@@ -132,33 +132,35 @@ func main() {
 // "/tile/0/x001/x234/067.p/8" means level 0, index 1234067 and width 8 of a partial tile.
 func parseTileLevelIndexWidth(level, index string) (uint64, uint64, uint64, error) {
 	l, err := strconv.ParseUint(level, 10, 64)
-	if err != nil {
+	if l > 63 || err != nil {
 		return 0, 0, 0, fmt.Errorf("failed to parse tile level")
 	}
 
-	var i, w uint64
-	switch indexPaths := strings.Split(index, "/"); len(indexPaths) {
-	// Full tile = 3
-	// Partial tile = 4
-	case 3, 4:
-		indexPath := strings.Join(indexPaths[0:3], "")
-		indexPath = strings.ReplaceAll(indexPath, "x", "")
-		indexPath = strings.ReplaceAll(indexPath, ".p", "")
-		i, err = strconv.ParseUint(indexPath, 10, 64)
-		if err != nil {
+	i := uint64(0)
+	w := uint64(256)
+	indexPaths := strings.Split(index, "/")
+
+	if strings.Contains(index, ".p") {
+		w, err = strconv.ParseUint(indexPaths[len(indexPaths)-1], 10, 64)
+		if err != nil || w < 1 || w > 255 {
 			return 0, 0, 0, fmt.Errorf("failed to parse tile index")
 		}
+		indexPaths[len(indexPaths)-2] = strings.TrimSuffix(indexPaths[len(indexPaths)-2], ".p")
+		indexPaths = indexPaths[:len(indexPaths)-1]
+	}
 
-		w = 256
-		if len(indexPaths) == 4 {
-			w, err = strconv.ParseUint(indexPaths[3], 10, 64)
-			if err != nil {
-				return 0, 0, 0, fmt.Errorf("failed to parse tile index")
-			}
-		}
-	default:
+	if strings.Count(index, "x") != len(indexPaths)-1 || strings.HasPrefix(indexPaths[len(indexPaths)-1], "x") {
 		return 0, 0, 0, fmt.Errorf("failed to parse tile index")
 	}
 
+	for _, indexPath := range indexPaths {
+		indexPath = strings.TrimPrefix(indexPath, "x")
+		n, err := strconv.ParseUint(indexPath, 10, 64)
+		if err != nil || n >= 1000 || len(indexPath) > 3 {
```

Comment:
Is the `len` part needed? The actual definition says that `It MUST be a non-negative integer encoded into zero-padded 3-digit path elements`, so this should actually be `len(indexPath) != 3`, but I think it's extra validation we don't strictly need? As in, if we can get a number out of it between [0,1000) then we're probably good to go.

Reply:
I do agree that it's an extra validation. The spec said it is a MUST, so that's why I added it here. Updated to `len(indexPath) != 3`.

Reply:
I'll go ahead to merge the PR. Let me know if you think that this should be removed.

Reply:
There are arguments both ways. I'm happy with this approach and it can be changed later if needed.